go.mod

```diff
@@ -21,6 +23,7 @@ require (
 	github.com/bufbuild/buf v0.20.5
 	github.com/buger/goterm v0.0.0-20180307092342-c9def0117b24
 	github.com/buger/jsonparser v0.0.0-20180808090653-f4dd9f5a6b44
+	github.com/chef/automate/api/external v0.0.0-00010101000000-000000000000
```

Comment:
Does this line need updated when changing the external protobufs?

Reply:
Is it going to be a two PR process to add an RPC to the external protobufs? One to check in the changes to the external protobufs then the second to update this line and add the code to the automate-gateway.

Reply:
Nope, that's why we ripped out the vendoring. The `replace github.com/chef/automate/api/external => ./api/external/` later in this file makes imports with `github.com/chef/automate/api/external` look in the local dir, and since we are not vendoring, that's enough to make it work. I explicitly tested by making changes to the applications service protos, compiling them and then referring to the updated stuff in the application service code. There's no need to commit or revendor or anything like that.

Reply:
I am trying the same kind of test that you mentioned. I added a new RPC to the external config-mgmt files. Then ran `compile_all_protobuf_components_nolint`. Then when building automate-gateway `build components/automate-gateway` got the below errors. 
```
y
# github.com/chef/automate/components/automate-gateway/handler
handler/cfgmgmt.go:140:66: undefined: "github.com/chef/automate/api/external/cfgmgmt/request".Errors2
handler/cfgmgmt.go:140:84: undefined: "github.com/chef/automate/api/external/cfgmgmt/response".Errors2
handler/cfgmgmt.go:146:10: undefined: "github.com/chef/automate/api/external/cfgmgmt/response".Errors2
   automate-gateway: Build time: 1m3s
   automate-gateway: Exiting on error
```
Below are the changes I made. 
https://github.com/chef/automate/commit/40dba9516b30081a6938e31b300b5febca272cfb

Reply:
@lancewf Your branch is based on an older version of this PR before I rebased on top of the latest mainline (with the novendor stuff merged). If I cherry pick your commit on top of the latest, there's one small bug in the `return` statement for the `GetErrors2` method, after fixing that, it builds correctly :)

Reply:
Ok thanks, I will try that out.

Reply:
That pulling the latest from this branch did fix the problem I was seeing, Sorry about that. I should have tried to update the code first.